iml-gui/crate/src/components/alert_indicator.rs

```diff
@@ -17,15 +19,20 @@ fn get_message(alerts: &[&Alert]) -> String {
 
 pub(crate) fn alert_indicator<T>(
     alerts: &HashMap<u32, Arc<Alert>>,
-    resource_uri: &str,
+    x: &dyn ToCompositeId,
     compact: bool,
     tt_placement: Placement,
 ) -> Node<T> {
+    let composite_id = x.composite_id();
+
     let alerts: Vec<&Alert> = alerts
         .arc_values()
-        .filter_map(|x: &Alert| match &x.affected {
-            Some(xs) => xs.iter().find(|x| x == &resource_uri).map(|_| x),
-            None => None,
+        .filter_map(|x| {
```

Comment:
I did a similar thing but with introducing `ResourceUri` trait. This is probably better.